### PR TITLE
merge changes from master to codeless

### PR DESCRIPTION
**Pull request by littleaj**
@trask I'm unsure about some of these. We'll see how the smoke tests fair. Let me know if anything looks off.

## Highlights
* w3c by default
* remove lf channel
* static analysis fixes

### CODE REVIEW

File: NOTICE [hunk 1]
```diff
@@ -32,21 +32,18 @@ Microsoft Application Insights Java Software Development Kit consists of materia
...
-19. Guava dependency: FindBugs JSR305 annotations 3.0.2 (https://github.com/findbugsproject/findbugs)
-20. Guava dependency: CheckerFramework annotations 2.5.2 (https://github.com/typetools/checker-framework/tree/master/checker-qual)
-21. Guava dependency: Google ErrorProne annotations 2.2.0 (https://github.com/google/error-prone/tree/master/annotations)
-22. Guava dependency: Google J2ObjC annotations 1.1 (https://github.com/google/j2objc/tree/master/annotations)
-23. Guava dependency: Animal Sniffer annotations 1.17 (https://github.com/mojohaus/animal-sniffer/tree/master/animal-sniffer-annotations)
-24. Glowroot Instrumentation 0.14.9 (https://github.com/glowroot/instrumentation)
-25. CheckerFramework annotations for Java 6 1.0.0 (https://github.com/trask/checker-framework-annotations-for-jdk6)
-26. Gson 2.8.5 (https://github.com/google/gson)
-27. os-maven-plugin 1.6.2 (https://github.com/trustin/os-maven-plugin)
+12. XStream version 1.4.11.1 (https://github.com/x-stream/xstream)
```
Comment (trask) [inline NOTICE]:
```suggestion
12. Moshi version 1.8.0 (https://github.com/square/moshi)
```

I forgot to do this when removing xstream in codeless branch

File: NOTICE [hunk 2]
```diff
@@ -791,396 +788,171 @@ END OF CREATIVE COMMONS ATTRIBUTION 4.0 INTERNATIONAL PUBLIC LICENSE
...
+ OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT
+ SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED
+ TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR
+ BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY
+ WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+ DAMAGE.
+ =========================================
+ END OF XStream NOTICES, INFORMATION, AND LICENSE
```
Comment (trask) [inline NOTICE]: Can you replace with this?
```
 %% Moshi NOTICES, INFORMATION, AND LICENSE BEGIN HERE
 =========================================
 Licensed under the Apache License, Version 2.0
 =========================================
 END OF Moshi NOTICES, INFORMATION, AND LICENSE
```
I forgot to do this when removing xstream in codeless branch

File: core/build.gradle [hunk 3]
```diff
@@ -26,31 +26,15 @@ apply plugin: 'idea'
```
Comment (trask) [inline core/build.gradle]: I think merge should remove the line above
```
apply plugin: 'idea'
```

Comment (littleaj) [reply core/build.gradle]: 👍

**Comment by littleaj** [comment]
@trask, I addressed the issues you raised and went ahead and merged the rest of the changes from master.

**Pull request merged**